lib/datadog/core/telemetry/collector.rb

```diff
@@ -0,0 +1,244 @@
+              compatible: integration_compatible?(integration),
+              error: (patch_error(integration) if is_instrumented && !is_enabled),
+              auto_enabled: is_enabled ? integration_auto_instrument?(integration) : nil
+            )
+          end
+        end
+
+        # Returns the runtime ID of the current process
+        def runtime_id
+          Datadog::Core::Environment::Identity.id
+        end
+
+        # Returns the current as a UNIX timestamp in seconds
+        def tracer_time
+          Time.now.to_i
+        end
+
+        private
+
+        TARGET_OPTIONS = [
+          'ci.enabled'.freeze,
+          'logger.level'.freeze,
+          'profiling.advanced.code_provenance_enabled'.freeze,
+          'profiling.advanced.endpoint.collection.enabled'.freeze,
+          'profiling.enabled'.freeze,
+          'runtime_metrics.enabled'.freeze,
+          'tracing.analytics.enabled'.freeze,
+          'tracing.distributed_tracing.propagation_inject_style'.freeze,
+          'tracing.distributed_tracing.propogation_extract_style'.freeze,
+          'tracing.enabled'.freeze,
+          'tracing.log_injection'.freeze,
+          'tracing.partial_flush.enabled'.freeze,
+          'tracing.partial_flush.min_spans_threshold'.freeze,
+          'tracing.priority_sampling'.freeze,
+          'tracing.report_hostname'.freeze,
+          'tracing.sampling.default_rate'.freeze,
+          'tracing.sampling.rate_limit'.freeze
+        ].freeze
+
+        def additional_payload_variables
+          # Whitelist of configuration options to send in additional payload object
+          config_options = Datadog.configuration.to_hash
+          config_options_to_keep = {}
+          TARGET_OPTIONS.each do |option|
+            config_options_to_keep[option] = format_configuration_value(config_options[option])
+          end
+
+          # Add some more custom additional payload values here
+          config_options_to_keep['tracing.auto_instrument.enabled'.freeze] = !defined?(Datadog::AutoInstrument::LOADED).nil?
+          config_options_to_keep['tracing.writer_options.buffer_size'.freeze] =
+            format_configuration_value(Datadog.configuration.tracing.writer_options[:buffer_size])
+          config_options_to_keep['tracing.writer_options.flush_interval'.freeze] =
+            format_configuration_value(Datadog.configuration.tracing.writer_options[:flush_interval])
+          config_options_to_keep['logger.instance'.freeze] = Datadog.configuration.logger.instance.class.to_s
+
+          compact_hash(config_options_to_keep)
+        end
+
+        def format_configuration_value(value)
+          # TODO: If the telemetry spec is updated to accept floats, this condition should be removed
+          if value.is_a?(Float) || value.is_a?(Array)
+            value.to_s
+          else
+            value
+          end
+        end
+
+        def compact_hash(hash)
+          hash.delete_if { |_k, v| v.nil? }
+        end
```

Comment:
This is the same as `Hash#compact!`: https://ruby-doc.org/core-2.4.0/Hash.html#method-i-compact-21